tests/backends/test_kafka.py

```diff
@@ -132,6 +134,35 @@ def test_auto_offset_reset_latest(self) -> None:
                 else:
                     raise AssertionError("expected EndOfPartition error")
 
+    def test_force_offset_reset_latest(self) -> None:
+        payload = KafkaPayload(b"a", b"0", [])
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                message = producer.produce(topic, payload).result(5.0)
+
+            # write a nonsense offset
+            with closing(self.get_consumer(auto_offset_reset="error")) as consumer:
+                consumer.subscribe([topic])
+                consumer.stage_positions(
+                    {
+                        message.partition: Position(
+                            offset=message.offset + 1000, timestamp=message.timestamp
+                        )
+                    },
+                    force=True,
```

Comment:
I would prefer not to introduce the `force` escape hatch. How about either doing `consumer.poll()` on the line before this one so `Consumer.__offsets` gets set, or waiting for the assignment callback to actually happen.

Reply:
I was unable to get this work with `poll()` which is why I had to do the force thing. I'm not sure if there is a good way to guarantee that the offsets are populated.

Reply:
@mitsuhiko I managed to get this to work with a combination of auto_offset_reset="earliest" on line 144 (so it can get an initial offset) and `consumer.poll(5.0)` on the line above this one. Sometimes it takes a while for the rebalance to happen but waiting up to 5 seconds did end up passing every time for me locally.

Reply:
I would prefer we just add the forcing behavior than to add another source of flakiness to tests.